db.js

```diff
@@ -105,157 +108,308 @@ function addWatch(user, item, price, server) {
+    }).catch(console.error)
+}
+
+function showBlocks(user, callback) {
+    return findOrAddUser(user)
+        .then((userId) => {
+            const queryStr = 'SELECT seller, server FROM blocked_seller_by_user WHERE user_id = $1'
+            connection.query(queryStr, [userId]).then((user_blocks) => {
+                const queryStr = 'SELECT seller, server, items.name, watches.server as item_server FROM blocked_seller_by_watch INNER JOIN watches ON blocked_seller_by_watch.watch_id = watches.id INNER JOIN items ON items.id = watches.item_id WHERE user_id = $1';
+                connection.query(queryStr, [userId]).then((watch_blocks) => {
+                    callback({user_blocks: user_blocks.rows, watch_blocks: watch_blocks.rows,})
+                })
+            })
+        }).catch(console.error)
+}
+
+function snooze(type, id, hours = 6) {
+    switch(type.toUpperCase()) {
+        case 'WATCH':
+            //insert into watch snoooze
+            (() => {
+                const queryStr = `INSERT INTO snooze_by_watch (watch_id, expiration) VALUES ($1, now() + interval '1 second' * $2) ON CONFLICT (watch_id) DO UPDATE SET expiration = now() + interval '1 second' * $2;`;
+                connection.query(queryStr, [id, hours * 60 * 60])
+                    .catch(console.error)})()
+            break;
+        case 'USER':
+            findOrAddUser(id)
+                .then((userId) => {
+                    //insert into account snooze
+                    (() => {
+                        const queryStr = `INSERT INTO snooze_by_user (user_id, expiration) VALUES ($1, now() + interval '1 hour' * $2) ON CONFLICT (user_id) DO UPDATE SET expiration = now() + interval '1 hour' * $2;`;
+                        connection.query(queryStr, [userId, hours])
+                            .catch(console.error)})();
+                })
+            break;
+    }
+}
+
+function unsnooze(type, id) {
+    switch(type.toUpperCase()) {
+        case 'WATCH':
+            (() => {
+                const queryStr = `DELETE FROM snooze_by_watch WHERE watch_id = $1;`;
+                connection.query(queryStr, [id])
+                    .catch(console.error)})()
+            break;
+        case 'USER':
+            findOrAddUser(id).then((userId) => {
+                (() => {
+                    const queryStr = `DELETE FROM snooze_by_user WHERE user_id = $1;`;
+                    connection.query(queryStr, [userId])
+                        .catch(console.error)})();
+            })
+            break;
+    }
+}
+
+function postSuccessfulCommunication(watchId, seller) {
+    const queryStr = `INSERT INTO communication_history (watch_id, seller, timestamp) VALUES ($1, $2, now()) ON CONFLICT ON CONSTRAINT communication_history_watch_id_seller_key DO UPDATE SET timestamp = now();`
+    connection.query(queryStr, [watchId, seller.toUpperCase()]).catch(console.error);
+}
+
+
+async function validateWatchNotification(userId, watchId, seller) {    
```

Comment:
I was trying really hard to get this all in one query with a bunch of joins, but wasn't able to get it working reliably.  Looking at it now I could just omit the else statements since the if statements always return- that would keep it from nesting horizontally.